### PR TITLE
Allow for custom 'add layout' menu components

### PR DESCRIPTION
**Pull request by happyDemon**
I personally needed an easier way to look for layouts I can add to a field.

I extracted the original implementation into its own component, kept it as default. 
I also added a search-select variant, it's use is documented in `docs/README.md`

### CODE REVIEW

**Comment by Nyratas** [comment]
Hi @happyDemon,

Thanks for your PR !

I am taking a look at it right now and will request some changes if needed.

Have great day.

File: resources/js/components/OriginalDropMenu.vue [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        <div class="relative" v-if="layouts.length > 1">
+            <div v-if="isLayoutsDropdownOpen"
+                 class="absolute rounded-lg shadow-lg max-w-full mb-3 pin-b max-h-search overflow-y-auto border border-40"
+            >
+                <div>
+                    <ul class="list-reset">
+                        <li v-for="layout in layouts" class="border-b border-40">
+                            <a @click="addGroup(layout)"
+                               class="cursor-pointer flex items-center hover:bg-30 block py-2 px-3 no-underline font-normal bg-20">
+                                <div><p class="text-90">{{ layout.title }} sdfsdf</p></div>
```
Comment (Nyratas) [inline resources/js/components/OriginalDropMenu.vue]: I guess there is a typo here ? I don't think the `sdfsdf` is meant to be here ?

File: resources/js/components/SearchMenu.vue [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+                this.$emit('addGroup', layout);
+
+                this.isLayoutsDropdownOpen = false;
+                this.selectedLayout = null;
+            },
+        }
+    }
+</script>
+<style src="vue-multiselect/dist/vue-multiselect.min.css"></style>
```
Comment (Nyratas) [inline resources/js/components/SearchMenu.vue]: Does this mean the new dropdown menu has its own style that does not match the "Nova" style ?

Comment (happyDemon) [reply resources/js/components/SearchMenu.vue]: it did, I extracted the sass files and placed it in `resources/sass` and configured it to be in line with general nova styling.

**Comment by happyDemon** [comment]
Here's how the search select menu looks:

Screenshot 2020-01-04 at 15 16 12

**Comment by Nyratas** [comment]
All right. Seems good to me. Thanks a lot!

Since a lot of people are waiting for a release with all the latest changes, I'll publish a release after merging your PR.

Have a great day!

**Pull request merged**